src/client/general/protocol.md

```diff
@@ -0,0 +1,58 @@
+# The Wayland protocol
+
+The Wayland protocol can be described as an asynchronous, object oriented protocol. This means clients and
+the compositor send and receive messages in order to build an abstract state machine made of objects.
+Wayland as a protocol is also asynchronous in the sense that a message will not complete immediately.
```

Comment:
This sentence feels weird, what does it mean for a message to "complete"?

I think I would rather insist on the fact that a large fraction of the Wayland messages are sent unprompted, and don't expect an answer, I think it better conveys the asynchronous aspect.